scripts/api_dump_generator.py

```diff
@@ -1769,11 +1795,20 @@ def __init__(self, rootNode, constants, aliases):
             self.namedParams = self.namedParams[0:-2]
             self.typedParams = self.typedParams[0:-2]
 
-        if self.parameters[0].type in ['VkInstance', 'VkPhysicalDevice'] or self.name == 'vkCreateInstance':
+        
+        if self.name in extensions and extensions[self.name].name == 'VK_EXT_debug_utils':
```

Comment:
I agree with you, but I am not confident that I don't catch other extensions by accident. Looking through the spec I didn't spot any, but one can never be too sure.

Reply:
One useful strategy is to compare the generated file before and after if you are concerned about other side effects (although api_dump currently makes this difficult because of its non-deterministic function order).

Also, the idea here would be to future-proof the codegen a little bit.  If a new instance extension with device functions comes along we want that to be dispatched properly too.